src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -436,6 +436,12 @@ public void RemoveNodeAddress(IPEndPoint ipEndpoint)
 
         public async Task<INetworkPeer> ConnectAsync(IPEndPoint ipEndpoint)
         {
+            if (this.connectedPeers.Any(connectedPeer => connectedPeer.PeerEndPoint.Match(ipEndpoint)))
+            {
+                this.logger.LogDebug("{0} is already connected.");
+                throw new Exception("{0} is already connected.");
```

Comment:
Does it have to throw an exception?
Can't it just return silently?

Reply:
I did think about this but then if we call this via the API we won't know what happened? We still need to refactor this to return a Json object or something so that the caller knows what happened. WDYT?

Reply:
Doe it matter? what if we just return the peer if its connected already
The caller then won't know if it was an "add" or an "already added" operation but does it matter?

Reply:
I think it doesn't matter. We can just return the already connected peer if we want.

Reply:
OK I've changed this to return 👍